subsys/net/lib/tftp/tftp_client.c

```diff
@@ -0,0 +1,524 @@
+/*
+ * Copyright (c) 2020 InnBlue
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(tftp_client, LOG_LEVEL_DBG);
+
+/* Public / Internal TFTP header file. */
+#include <net/tftp.h>
+#include "tftp_client.h"
+
+/* TFTP Request Buffer / Size variables. */
+static u8_t   tftpc_request_buffer[TFTPC_MAX_BUF_SIZE];
+static u32_t  tftpc_request_size;
+
+/* TFTP Request block number property. */
+static u32_t  tftpc_block_no;
+static u32_t  tftpc_index;
+
+/* Global mutex to protect critical resources. */
+K_MUTEX_DEFINE(tftpc_lock);
+
+/* Default TFTP Server (the user can provide an override via API). */
+static struct sockaddr_in def_srv = {
+	.sin_family = AF_INET,
+	.sin_port   = htons(TFTPC_DEF_SERVER_PORT)
+};
+
+/* Global timeval structure indicating the timeout interval
+ * for all TFTP transactions.
+ */
+static struct timeval tftpc_timeout = {
+	.tv_sec   =   CONFIG_TFTPC_REQUEST_TIMEOUT,
+	.tv_usec  =   0
+};
+
+/* Number of simultaneous client connections will be
+ * NUM_FDS be minus 2
+ */
+fd_set readfds;
+fd_set workfds;
```

Comment:
Unused variable?

Reply:
All remaining variables should be `static` - this helps to identify such cases easily.

Reply:
Fixed..